src/apps/rack_level_slots/routers.py

```diff
@@ -0,0 +1,143 @@
+from typing import Union
+
+from fastapi import Depends, Request, Response, status
+from fastapi.responses import JSONResponse
+from fastapi.routing import APIRouter
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from src.apps.rack_level_slots.schemas import (
+    RackLevelSlotBaseOutputSchema,
+    RackLevelSlotInputSchema,
+    RackLevelSlotOutputSchema,
+    RackLevelSlotUpdateSchema,
+)
+from src.apps.rack_level_slots.services import (
+    activate_single_rack_level_slot,
+    add_single_stock_to_rack_level_slot,
+    create_rack_level_slot,
+    deactivate_single_rack_level_slot,
+    get_all_rack_level_slots,
+    get_single_rack_level_slot,
+    update_single_rack_level_slot,
+)
+from src.apps.stocks.schemas.stock_schemas import StockRackLevelSlotInputSchema
+from src.apps.users.models import User
+from src.core.pagination.models import PageParams
+from src.core.pagination.schemas import PagedResponseSchema
+from src.core.permissions import check_if_staff, check_if_staff_or_has_permission
+from src.dependencies.get_db import get_db
+from src.dependencies.user import authenticate_user
+
+rack_level_slot_router = APIRouter(prefix="/rack-level-slots", tags=["rack_level_slot"])
+
+
+@rack_level_slot_router.post(
+    "/",
+    response_model=RackLevelSlotOutputSchema,
+    status_code=status.HTTP_201_CREATED,
+)
+async def post_rack_level_slot(
+    rack_level_slot: RackLevelSlotInputSchema,
+    session: AsyncSession = Depends(get_db),
+    request_user: User = Depends(authenticate_user),
+) -> RackLevelSlotOutputSchema:
+    await check_if_staff(request_user)
+    return await create_rack_level_slot(session, rack_level_slot)
```

Comment:
this endpoint is useless - rack level slots are managed automatically when rack level slots are remove oraz created when the max slots amount changes
delete it